DSCResources/MSFT_xDSCWebService/MSFT_xDSCWebService.psm1

```diff
@@ -291,19 +298,35 @@ function Set-TargetResource
 
     PSWSIISEndpoint\Set-AppSettingsInWebconfig -path $PhysicalPath -key "RegistrationKeyPath" -value $RegistrationKeyPath
 
+    $iisSelfSignedModuleAssemblyName = "IISSelfSignedCertModule.dll"
+    $iisSelfSignedModuleName = "IISSelfSignedCertModule(32bit)"
     if($AcceptSelfSignedCertificates)
-    {
-        Copy-Item "$pathPullServer\IISSelfSignedCertModule.dll" $env:windir\System32\inetsrv -Force
-        Copy-Item "$env:windir\SysWOW64\WindowsPowerShell\v1.0\Modules\PSDesiredStateConfiguration\PullServer\IISSelfSignedCertModule.dll" $env:windir\SysWOW64\inetsrv -Force
+    {        
+        $preConditionBitnessArgumentFor32BitInstall=""
+        if ($Enable32BitAppOnWin64 -eq $true)
```

Comment:
What if we are on a 32 bit machine? Won't this break now if I am already using this resource on a 32 bit machine and don't have this parameter in my script?

Reply:
The Pull Server is only valid for a 64 bit system. The flag is to tell the resource whether the Pull Server needs to run as a 32 bit or a 64 bit process. Let me know if this answers your question. 

---
In reply to: [147538235](https://github.com/PowerShell/xPSDesiredStateConfiguration/pull/372#discussion_r147538235) [](ancestors = 147538235)

Reply:
It looks like previously, we always ran a 32 bit process by default, but now we run a 64 bit process by default and only run a 32 bit process if you specify a parameter that didn't exist in previous versions. So it's a breaking change because if you upgrade to this version, your process will now be 64 bit instead of 32 bit. Since you are removing the precondition 32 bit by default. Is this an expected breaking change?

Also, shouldn't $iisSelfSignedModuleName be the 64 bit name of the module by default and then set to the 32 bit name if you want the 32 bit version? Right now it looks like the 32 bit version is always used.

Reply:
I feel this is an acceptable breaking change for the following reasons
1) This does not break DSC functionality in any way
2) This actually fixes some of the load problems people have been seeing with ESENT when DSC Pull Server is hosted in a 32 bit process. 

For your other question of using the 32 bit module name - in my first iteration, I had different module names. But that just made the code uglier and also did not handle the case when the user has pull server running in 32 bit and then wants to upgrade it to run in 64 bit by updating the property in IIS manager. For that scenario to work (which a lot of our customers are using), the module name had to be the same for both. Hence, I just used the 32 bit in the module name as that's what existed in the older version of the resource.

Reply:
Indhu and I talked about this offline. This is a change in default behavior so I am labeling it as a breaking change. But this is the correct behavior now and it should not actually break anyone's scripts.